### ISSUE TITLE
How to debug OSv application

### ISSUE THREAD

**Issue by justinc1** [cloudius-systems/osv#662]
If I wait long enough that application (orted.so) is loaded and run, I can do
```
(gdb) hb orte_daemon_recv
Hardware assisted breakpoint 2 at 0x10000543e133: file ../../orte/orted/orted_comm.c, line 92.
```

But I want to catch very first invocations of orte_daemon_recv, so I try to set breakpoint early, while orted.so is not loaded yet.
```
(gdb) hb orte_daemon_recv
Function "orte_daemon_recv" not defined.
Make hw breakpoint pending on future shared library load? (y or [n]) y
Hardware assisted breakpoint 1 (orte_daemon_recv) pending.
(gdb) c
```

Now target OSv container continues, orte_daemon_recv is called - it prints some debug output. But breakpoint is newer triggered. Why? And how to make it happen?

I guess it might help to press Ctrl+C in gdb, after .so is loaded, and before interesting code is executed, and do "osv syms" and "hb ..." at that point? But others might have better solution.

**Comment by justinc1** [comment]
With delay added to orted.so, so that "hb ..." is executed while orted.so is already running:
```
(gdb) hb orte_daemon_recv
Cannot access memory at address 0x10000543e133
```
orted.so at orted.so 0x100005400000, .text section at 0x10000543af30, and nm telling "000000000003e133 T orte_daemon_recv", looks ok. 

So why "cannot access memory"?

**Comment by nyh** [comment]
Hi Justin, I'm closing this issue, which is more a discussion than a specific problem, and we can continue to discuss it on the mailing list - and when we come up with a more precise problem we want to fix, you can open a new issue or issues. Thanks!
